ui/src/contexts/APIProvider.tsx

```diff
@@ -59,6 +68,7 @@ export function APIProvider({ children, autoStartAPIEvents = true }: APIProvider
   const [upstreamLogs, setUpstreamLogs] = useState("");
   const [metrics, setMetrics] = useState<Metrics[]>([]);
   const [connectionStatus, setConnectionState] = useState<ConnectionState>("disconnected");
+  const [versionInfo, setVersionInfo] = useState<VersionInfo | null>(null);
```

Comment:
I think we can remove the getVersionInfo() function and replace it with useEffect that runs once on init: 

```
useEffect(() => {
    // fetch version 
   try {
      const response = await fetch("/api/version");
      if (!response.ok) {
        throw new Error(`HTTP error! status: ${response.status}`);
      }
      const data: VersionInfo = await response.json();
      setVersionInfo(data);
  } catch (error) {
      console.error(error);
  }
}, []);
```